Openvstorage/Framework documentation/django_stack.md

```diff
@@ -0,0 +1,49 @@
+
+
+## Django stack
+As could be expected, our django stack is located in the `webapps.api` folder. We only use a subset of the whole Django functionality.
+We do use their routing mechanism, and their views to implement these. Serialization, however, from api response values is overridden by us, as Django cannot cope too well with our DAL structure. We don't use their ORM for this reason.
+### Default Django files and contents
+- wsgi.py
+- urls.py: implementation of the Django simplerouter. An own implementation is needed to backport features introduced in newer django versions.
+implements static and dynamic list and detail routes, and functions to access these.
+The default django `urlpatterns` variable contains routes for the built views:
+
+
+|routes  | views  |
+|---|---|
+| ovs api calls  |  `def build_router_urls` |
+| authentication, identification, plugin metadata |  `metadata.py`|
+| openAPI according to swagger| `openapi.py`|
+| relay | `view.py`, `def relay` |
+| oath2 token creation and redirection | `oath2` module |
+The `PluginController.get_webapps` fetches all viewclasses in `api/backend/views`.
+
+All ovs implemented routes in this module are registered through the base django rest framework router. 
+   
+- settings.py
+- manage.py
+
+## Frontend stack
+
```

Comment:
Frontend relies on the Durandal framework, which uses Knockout for databinding.
Link to durandal: http://durandaljs.com

Durandal is a MVC framework meaning each view has an associated viewmodel. This is reflected in the structure of the application.
The JS version that we opt for is ES4 with certain ES5 backported in the polyfill.